bot/cogs/stats.py

```diff
@@ -0,0 +1,65 @@
+from discord import Member, Message, Status
+from discord.ext.commands import Bot, Cog, Context
+
+
+class Stats(Cog):
+    """A cog which provides a way to hook onto Discord events and forward to stats."""
+
+    def __init__(self, bot: Bot):
+        self.bot = bot
+
+    @Cog.listener()
+    async def on_message(self, message: Message) -> None:
+        """Report message events in the server to statsd."""
+        if message.guild is None:
+            return
+
+        reformatted_name = message.channel.name.replace('-', '_')
+
+        if reformatted_name.startswith("ot"):
+            # Off-topic channels change names, we don't want this for stats.
+            # This will change 'ot1-lemon-in-the-dishwasher' to just 'ot1'
+            reformatted_name = reformatted_name[:3]
+
+        stat_name = f"channels.{reformatted_name}"
+        self.bot.stats.incr(stat_name)
+
+        # Increment the total message count
+        self.bot.stats.incr("messages")
+
+    @Cog.listener()
+    async def on_command_completion(self, ctx: Context) -> None:
+        """Report completed commands to statsd."""
+        command_name = ctx.command.qualified_name.replace(" ", "_")
+
+        self.bot.stats.incr(f"commands.{command_name}")
+
+    @Cog.listener()
+    async def on_member_join(self, member: Member) -> None:
+        """Update member count stat on member join."""
+        self.bot.stats.gauge(f"guild.total_members", len(member.guild.members))
```

Comment:
The bot is in some other guilds, like the emoji server. If not accounted for, the total could be overwritten. I think all events should explicitly check for the main guild.